src/index.tsx

```diff
@@ -1,16 +1,55 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import {
+  createBrowserRouter,
+  RouterProvider,
+  redirect
+} from "react-router-dom";
 import "./index.css";
 import App from "./App";
 import logo from "./logo.svg";
 
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+    loader: ({ params }) => {
+      // If the app was accessed through old URL format, redirect
+      const txhash = new URLSearchParams(window.location.search).get("t");
+
+      // This loader is executed when accessing tx/:txhash, we need this check to avoid infinite loop
+      if (txhash?.length === 66 && !params.txhash) {
+        return redirect(`tx/${txhash}`);
+      }
+    },
+    children: [
+      {
+        path: "tx",
+        element: null,
+        loader: ({ params }) => {
+          const { txhash } = params;
+          if (!txhash || txhash.length !== 66) {
+            return redirect("/");
+          }
+        },
+        children: [
+          {
+            path: "/tx/:txhash",
+            element: <App />
+          }
```

Comment:
While testing out a wrong route `//tx/:txhash`, I get this router error UX. Are we sure this won't come in prod?

<img width="1714" alt="image" src="https://user-images.githubusercontent.com/7558499/204728520-90d1d66e-f912-41e5-9d5d-fa06a71dd8a1.png">

URL : http://localhost:3000//tx/0x471094aee2d350b98927d2f6092a284b5092a83496cf6f3681db091c27062fd8

Reply:
AFAIK it should not happen in production

Reply:
you can test on the vercel deployment as it's using the production env
https://retryable-tx-panel-git-add-react-router-offchain-labs.vercel.app/
the duplicate / is immediately stripped away so it is fine

Reply:
Can we go with an implementation that works in both envs?

Reply:
I think the reason it works in production is because of vercel hosting itself, not because of our config.

https://retryable-tx-panel.arbitrum.io//tx redirect properly to https://retryable-tx-panel.arbitrum.io/tx (without router)